docs/articles/documentation/extending-testcafe/custom-reporter/helper-methods.md

```diff
@@ -0,0 +1,328 @@
+---
+layout: docs
+title: Helper Methods
+permalink: /documentation/extending-testcafe/custom-reporter-plugin/helper-methods
+---
+# Helper Methods
+
+The helper methods are used while [implementing a reporter](creating-reporter-plugin.md#implementing-reporter)
+to format a report output. TestCafe mixins these methods to the reporter.
+
+To refer to the helper methods, use `this`.
+
+## Output Helpers
+
+All output helper methods are chainable: each method returns `this` so that the methods can be executed one right after the other.
+This allows you to write compact code like this:
+
+```js
+this.setIndent(4)
+    .write(testname)
+    .newline()
+    .write(result);
+```
+
+### newline
+
+Adds a new line to the report.
+
+```text
+newline () → this
+```
+
+**Example**
+
+```js
+reportTaskDone (endTime, passed) {
+    ...
+    var summary = passed === this.testCount ?
```

Comment:
In all code snippets: we don't use `var`. We use `const` for immutable variables or `let` for the muttable